### ISSUE TITLE
Database migration fails if collection has an TTL index

### ISSUE THREAD

**Issue by hmsantos** [Hangfire-Mongo/Hangfire.Mongo#278]
Hi,

When we tried to upgrade to the latest Hangfire.Mongo version in one of our projects, on startup we get an  InvalidCastException (Unable to cast object of type 'MongoDB.Bson.BsonDouble' to type 'MongoDB.Bson.BsonInt64'). This only happens in one project where we have added a TTL index to the jobGraph collection and that is causing the database migration to fail. 
The exception is being thrown by this line: https://github.com/sergeyzwezdin/Hangfire.Mongo/blob/7c3ae992719a0747d71380f13c97af1ef0769660/src/Hangfire.Mongo/Migration/Strategies/Backup/CollectionMongoBackupStrategy.cs#L105
We tried running the backup strategy code in a simple console app and in fact the BsonType for the "expireAfterSeconds" being return is Double thus causing the exception
![image](https://user-images.githubusercontent.com/1530895/116415592-797c8680-a831-11eb-9de3-9ed51ae38b44.png)

**Comment by gottscj** [comment]
@hmsantos,

I added a check for the type.
Could you elaborate on what you are trying to fix with the TTL index? Hangfire.Mongo comes with a "MongoExpirationManager" which deletes expired records. Is something not working correctly?

**Comment by hmsantos** [comment]
@gottscj 
Thanks for adding the check.
We added the TTL index because for some reason in that service the jogGraph documents expiration date isn't always being set and because of that the database kept growing. We didn't find any pattern of which documents failed to set the expiration date so we created the index to control the database size and upgraded to the latest Hangfire.Mongo version at the time. Will try to have another look when possible
